src/wp-includes/rest-api/endpoints/class-wp-rest-block-types-controller.php

```diff
@@ -705,6 +706,19 @@ public function get_item_schema() {
 				),
 				'keywords'              => $keywords_definition,
 				'example'               => $example_definition,
+				'block_hooks'           => array(
+					'description'       => __( 'This block is automatically inserted near any occurence of the block types used as keys of this map, into a relative position given by the corresponding value.' ),
+					'type'              => 'object',
+					'patternProperties' => array(
+						'^[a-zA-Z0-9-]+/[a-zA-Z0-9-]+$' => array(
```

Comment:
@ockham How something like this.

```
$properties = array();
$registry   = WP_Block_Type_Registry::get_instance();
foreach( $registry->get_all_registered() as $block ){
   $properties[ $block ] = array(
       'type' => 'string',
       'enum' => array( 'before', 'after', 'first_child', 'last_child' ),
   );
}

Reply:
Noting that there might be hundreds of registered blocks, but the block author might still list the block type name that isn't present on the site and we would consider it valid. What's the reasoning behind listing all available blocks?